src/event-processor/event-processor.service.ts

```diff
@@ -0,0 +1,223 @@
+import {
+  Injectable,
+  Logger,
+  OnModuleInit,
+  OnApplicationShutdown,
+} from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { Cron } from '@nestjs/schedule';
+import * as amqp from 'amqplib';
+import { ConnectorService } from './connector.service';
+import { UpdatePaymentStatusDto } from './dto/update-payment-status.dto';
+import { PaymentStatus } from 'src/shared/enums/payment-status.enum';
+import { ShipmentStatus } from 'src/shared/enums/shipment-status.enum';
+import { UpdateShipmentStatusDto } from './dto/update-shipment-status';
+// import { ConnectionService } from './connection.service'; // Adjust import path as needed
+
+/**
+ * The service to process events from the respective queues.
+ * @property processingTime The upper limit of simulated processing time
+ * @property connection The current RabbitMQ connection
+ * @property queues An array containing all queue identifiers, to connect to
+ * @property maxMessages The rate limit for each queue per minute
+ * @property messageCounts The current message count
+ * @property booolean flag
+ */
+@Injectable()
+export class EventProcessorService
+  implements OnModuleInit, OnApplicationShutdown
+{
+  private readonly processingTime: number;
+  private connection: amqp.Connection;
+  private channel: amqp.Channel;
+  private queues = ['payments-queue', 'shipments-queue'];
+  private maxMessages: number;
+  private successRate: number;
+  private messageCounts: { [key: string]: number } = {};
+  private processingAllowed: { [key: string]: boolean } = {};
+
+  constructor(
+    private readonly logger: Logger,
+    private configService: ConfigService,
+    private connectorService: ConnectorService,
+  ) {
+    // read configuration from environment variables
```

Comment:
long term this should be done using the experiment config stuff, but until this exists, this is fine for now

Reply:
Event then, it probably makes sense to have a default value. Since it needs to use something, until the event returns from experiment-config with the configured variables